test/libyul/yulOptimizerTests/loopInvariantCodeMotion/no_move_state.yul

```diff
@@ -0,0 +1,40 @@
+{
+  let b := 1
+  for { let a := 1 } iszero(eq(a, 10)) {sstore(0x00, 0x00)} {
+    let inv := add(b, 42)
+    let x := extcodesize(mul(inv, 3))
+    a := add(x, 1)
+    mstore(a, inv)
+  }
+
+  for { let a := 1 } iszero(eq(a, 10)) { a := add(a, 1) } {
+    let inv := add(b, 42)
+    pop(create(0x08, 0x00, 0x02)) // invalidates state
+    let x := sload(mul(inv, 3))
+    a := add(x, 1)
+    mstore(a, inv)
+  }
+}
```

Comment:
Can you add two more loops where the state is invalidated in the pre-part and in the condition, respectively?

Reply:
The `ForLoopInitRewriter` is run before `LoopInvariantCodeMotion.` So the test for the pre part should not make any difference.

Reply:
And it does assert that the init is empty, right? (otherwise you would also swap some instructions)